### PR TITLE
how to listen when animation ends? #36

### PR DESCRIPTION
**Pull request by lovejjfg**
add enterListener and exitListener in `Transition` !

```
    exitTransition = ActivityTransition
            .with(intent)
            .to(findViewById(R.id.sub_imageView))
            .interpolator(new BounceInterpolator())
            .enterListener(new SimpleAnimatorListener() {
                @Override
                public void onAnimationStart(Animator animation) {
                    Log.e("TAG", "onEnterAnimationStart: ");
                }

                @Override
                public void onAnimationEnd(Animator animation) {
                    Log.e("TAG", "onEnterAnimationEnd: ");
                }

            })
            .start(savedInstanceState)
            .exitListener(new SimpleAnimatorListener() {
                @Override
                public void onAnimationStart(Animator animation) {
                    Log.e("TAG", "onOutAnimationStart: ");
                }

                @Override
                public void onAnimationEnd(Animator animation) {
                    Log.e("TAG", "onOutAnimationEnd: ");
                }
            });
```

### CODE REVIEW

**Comment by takahirom** [comment]
👀

**Comment by takahirom** [comment]
I fixed ci build.
Can you rebase master?

Like this

```
 git pull --rebase git@github.com:takahirom/PreLollipopTransition.git master
```

**Review by takahirom — CHANGES_REQUESTED** [review]
Sorry. Please change dependency for ci build.

https://github.com/takahirom/PreLollipopTransition/blob/master/sample-app/build.gradle#L67

``` groovy
    compile project(":pre-lollipop-activity-transition")

//    compile('com.kogitune:pre-lollipop-activity-transition:1.3.2') {
//        exclude module: 'support-v4'
//    }
```

**Comment by lovejjfg** [comment]
Should i change to use this`compile project(":pre-lollipop-activity-transition")`?   before i commit  ,there is used the following dependency...

**Comment by takahirom** [comment]
@lovejjfg 
Yes.. It is my mistake. Sorry. 🙇 
I want to confirm ci build.

**Review by takahirom — APPROVED** [review]

File: pre-lollipop-activity-transition/src/main/java/com/kogitune/activity_transition/SimpleAnimatorListener.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.kogitune.activity_transition;
+
+import android.animation.Animator;
+
+/**
+ * Created by Joe on 2016/10/8.
+ * Email lovejjfg@gmail.com
+ */
+
+public abstract class SimpleAnimatorListener implements Animator.AnimatorListener {
```
Comment (takahirom) [inline pre-lollipop-activity-transition/src/main/java/com/kogitune/activity_transition/SimpleAnimatorListener.java]: Probably you can use AnimatorListenerAdapter.
https://developer.android.com/reference/android/animation/AnimatorListenerAdapter.html

File: sample-app/src/main/java/com/kogitune/prelollipoptransition/support_fragment/SupportStartFragment.java [hunk 2]
```diff
@@ -34,7 +34,10 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             @Override
             public void onClick(View view) {
                 final SupportEndFragment toFragment = new SupportEndFragment();
-                FragmentTransitionLauncher.with(view.getContext()).from(view.findViewById(R.id.support_fragment_start_imageview)).prepare(toFragment);
+                FragmentTransitionLauncher
```
Comment (takahirom) [inline sample-app/src/main/java/com/kogitune/prelollipoptransition/support_fragment/SupportStartFragment.java:37]: Thank you for readable code! 👍

File: sample-app/src/main/java/com/kogitune/prelollipoptransition/SubActivity.java [hunk 3]
```diff
@@ -32,7 +35,42 @@
...
+                    }
+
+                    @Override
+                    public void onAnimationStart(Animator animation) {
+                        Log.e("TAG", "onOEnterAnimationStart ");
+                    }
+                })
+                .to(findViewById(R.id.sub_imageView))
+                .start(savedInstanceState)
+                .exitListener(new Animator.AnimatorListener() {
```
Comment (takahirom) [inline sample-app/src/main/java/com/kogitune/prelollipoptransition/SubActivity.java]: Probably If you split code, code become more readable.

``` java
        exitTransition = ActivityTransition
                .with(getIntent())
                .enterListener(new SimpleAnimatorListener() {
                    @Override
                    public void onAnimationEnd(Animator animation) {
                        Log.e("TAG", "onEnterAnimationEnd");
                    }

                    @Override
                    public void onAnimationStart(Animator animation) {
                        Log.e("TAG", "onOEnterAnimationStart ");
                    }
                })
                .to(findViewById(R.id.sub_imageView))
                .start(savedInstanceState);

       exitTransition.exitListener(new Animator.AnimatorListener() {
                    @Override
                    public void onAnimationStart(Animator animation) {
                        Log.e("TAG", "onOutAnimationStart: ");
                    }

                    @Override
                    public void onAnimationEnd(Animator animation) {
                        Log.e("TAG", "onOutAnimationEnd: ");
                    }

                    @Override
                    public void onAnimationCancel(Animator animation) {

                    }

                    @Override
                    public void onAnimationRepeat(Animator animation) {

                    }
                });
```

**Review by takahirom — CHANGES_REQUESTED** [review]
Please modify some code. if you have question please let me know

File: pre-lollipop-activity-transition/src/main/java/com/kogitune/activity_transition/SimpleAnimatorListener.java (hunk 1 above)
Comment (lovejjfg) [reply pre-lollipop-activity-transition/src/main/java/com/kogitune/activity_transition/SimpleAnimatorListener.java]: oh It's my fault...

**Review by takahirom — APPROVED** [review]

**Comment by takahirom** [comment]
Merged

**Comment by thanhbinh84** [comment]
the version 1.3.2 doesn't include this code. could you add a new release for this. Thanks.

**Pull request merged**